app/validation/validator.py

```diff
@@ -24,23 +23,39 @@ def validate_schema(self, json_to_validate):
 
         errors = []
 
-        errors.extend(self.validate_schema_contains_valid_routing_rules(json_to_validate))
+        numeric_answer_ranges = {}
 
-        errors.extend(self.validate_calculated_ids_in_answers_to_calculate_exists(json_to_validate))
+        for section in json_to_validate['sections']:
+            for group in section['groups']:
 
-        errors.extend(self.validate_routing_rules_has_default_if_not_all_answers_routed(json_to_validate))
+                if section == json_to_validate['sections'][-1]:
+                    if group == section['groups'][-1]:
+                        errors.extend(self.validate_contains_confirmation_or_summary(group['blocks']))
 
-        errors.extend(self.validate_numeric_answer_types(json_to_validate))
+                for block in group['blocks']:
+
+                    if 'routing_rules' in block and block['routing_rules']:
+                        for rule in block['routing_rules']:
+                            errors.extend(self.validate_schema_contains_valid_routing_rules(section['groups'], group['blocks'], rule))
```

Comment:
I know it never has but shouldn't we be validating routing rules at a group level as well?

Reply:
Just to further complicate this. Should this be extended for skip conditions?

Reply:
added group routing

Reply:
skip conditions use when not goto